Prose/ProseLib/Sources/JoinChatFeature/AddMemberSheet.swift

```diff
@@ -0,0 +1,256 @@
+//
+// This file is part of prose-app-macos.
+// Copyright (c) 2022 Prose Foundation
+//
+
+import AppLocalization
+import Assets
+import ComposableArchitecture
+import ProseCoreTCA
+import ProseUI
+import SwiftUI
+
+private let l10n = L10n.AddMember.self
+
+// MARK: - View
+
+public struct AddMemberSheet: View {
+  public typealias ViewState = AddMemberSheetState
+  public typealias ViewAction = AddMemberSheetAction
+
+  let store: Store<ViewState, ViewAction>
+
+  public init(store: Store<ViewState, ViewAction>) {
+    self.store = store
+  }
+
+  public var body: some View {
+    WithViewStore(self.store) { viewStore in
+      VStack(spacing: 16) {
+        VStack(alignment: .leading) {
+          Text(verbatim: l10n.Header.title)
+            .font(.headline)
+          Text(verbatim: l10n.Header.subtitle)
+            .font(.subheadline)
+        }
+        .fixedSize(horizontal: false, vertical: true)
+        .frame(maxWidth: .infinity, alignment: .leading)
+        // TODO: [Rémi Bardon] Refactor `BasicAuthView.chatAddressHelpButton` so we can add the button here too.
+        // TODO: [Rémi Bardon] Maybe we could even create a `JIDField`, which supports all of this out of the box.
+        TextField(
+          l10n.TextField.label,
+          text: viewStore.binding(\.$text),
+          prompt: Text(verbatim: l10n.TextField.prompt)
+        )
+        .textFieldStyle(.roundedBorder)
+        .onSubmit(of: .text) { viewStore.send(.submitTapped) }
+        .frame(maxWidth: .infinity, alignment: .leading)
+        Self.infoMessage(for: viewStore.info)
+          .symbolVariant(.fill)
+          .fixedSize(horizontal: false, vertical: true)
+          .redacted(reason: viewStore.isProcessing ? .placeholder : [])
```

Comment:
When `isProcessing` is true for the first time you don't see any activity indicator as a user. Redacting also doesn't strike me as a particular great way to communicate progress in this use case. I'd prefer a progress indicator in the submit button for example.

Reply:
I added a spinner before the buttons and removed the `.redacted`:
<img width="472" alt="Screenshot 2022-07-26 at 11 35 29" src="https://user-images.githubusercontent.com/37386490/180974874-fbbf7a47-776e-4bde-aa01-f9fc98c1e07f.png">